### PR TITLE
refactor: jira plugin

### PR DESCRIPTION
**Pull request by steinliber**
Signed-off-by: Meng JiaFeng <jiafeng.meng@merico.dev>

## Pre-Checklist

Note: please complete **_ALL_** items in the following checklist.

- [x] I have read through the [CONTRIBUTING.md](https://github.com/devstream-io/devstream/blob/main/CONTRIBUTING.md) documentation.
- [x] My code has the necessary comments and documentation (if needed).
- [x] I have added relevant tests

## Description
- rename `jira-github-integ` to `jira` plugin, because we can support different scm type

## Related Issues
#1336

### CODE REVIEW

File: docs/plugins/plugins-list.zh.md [hunk 1]
```diff
@@ -5,7 +5,7 @@
 | ---------------------- | ------------------- | ------------------------------- | --------------------------------------- |
 | Issue Tracking         | trello-github-integ | Trello/GitHub 整合              | [doc](trello-github-integ.md)           |
 | Issue Tracking         | trello              | Trello 配置                     | [doc](trello.md)                        |
-| Issue Tracking         | jira-github-integ   | Jira/GitHub 整合                | [doc](jira-github-integ.md)             |
```
Comment (daniel-hutao) [inline docs/plugins/plugins-list.zh.md:8]: “Jira/GitHub 整合” should be updated accordingly.

File: internal/pkg/plugin/jira/tpl/issuebuilder.tpl.yml [hunk 2]
```diff
@@ -1,6 +1,4 @@
-package jiragithub
```
Comment (daniel-hutao) [inline internal/pkg/plugin/jira/tpl/issuebuilder.tpl.yml:1]: Why still use `jira-github-integ` here?

File: internal/pkg/show/config/plugins/jira.yaml [hunk 3]
```diff
@@ -0,0 +1,27 @@
+tools:
+# name of the tool
+- name: jira-github-integ
```
Comment (daniel-hutao) [inline internal/pkg/show/config/plugins/jira.yaml:3]: need to be updated.

**Review by daniel-hutao — CHANGES_REQUESTED** [review]

**Comment by daniel-hutao** [comment]
@steinliber Please resolve the conflicts in this pr.

**Review by daniel-hutao — APPROVED** [review]

**Review by IronCore864 — APPROVED** [review]

**Pull request merged**